### PR TITLE
ARROW-18227: [CI][Packaging] Do not fail conda-clean if conda search raises PackagesNotFound

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit conda-clean

**Review by assignUser — APPROVED** [review]
LGTM `722 builds may be deleted out of 1107` :exploding_head: 

There is a linter error though: https://github.com/apache/arrow/actions/runs/3377501672/jobs/5606508249#step:5:830

**Comment by raulcd** [comment]
@pitrou what do you think about this approach to avoid the current failures on our conda-clean job?

File: dev/tasks/conda-recipes/clean.py [hunk 1]
```diff
@@ -21,37 +21,56 @@
...
+                platform
+            ],
+        )
+    except CommandFailedException as ex:
+        # If the command failed due to no packages found, return
+        # 0 builds to delete.
+        if "PackagesNotFoundError" in ex.output:
+            return 0
+        else:
+            sys.exit(1)
```
[the remark this answers is not in the document]
Comment (pitrou) [reply dev/tasks/conda-recipes/clean.py:72]: Ah, I see, it was already exiting before.

**Review by pitrou — APPROVED** [review]

**Pull request merged**